### DISCUSSION TITLE
Why YimMenu doesn't have recovery or toxic features?

### DISCUSSION THREAD

**Discussion by Yimura** [YimMenu/YimMenu#237] [Announcements]
### The main vision of YimMenu is patching crashes and making the overall experience of GTA V better.
In this vision there's no place for recovery and toxic features.

## Why?
Because nobody likes being griefed by modders and YimMenu doesn't like being griefed by Take2.

**Comment by maybegreat48** [comment]
A good feature would be to automatically protect grinders (detect business source/sell) somehow. Give them OTR, no cops, semi godmode (through looping pickup rewards), vehicle godmode etc.

```c++
using give_pickup_rewards = void (*) (int players, uint32_t hash);
main_batch.add("GPR", "48 89 5C 24 ? 57 48 83 EC 20 8B F9 48 8B 0D ? ? ? ? 8B DA B2 01 E8 ? ? ? ? 48 8B 0D ? ? ? ? E8 ? ? ? ? 48 8B C8 33 C0 48 85 C9 74 0A 44 8B C3 8B D7 E8 ? ? ? ? 48 8B 0D ? ? ? ? 48 8B D0 48 8B 5C 24 ? 48 83 C4 20 5F E9 ? ? ? ?", [this](memory::handle ptr)
{
	m_give_pickup_rewards = ptr.as<functions::give_pickup_rewards>();
});

namespace big
{
	void looped::player_good_options()
	{
		for (auto& player : g_player_service->m_players)
		{
			if (player.second->m_semi_godmode) 
			{
				if (CPed* ped = player.second->get_ped()) 
				{
					if (ped->m_maxhealth == 0 || ped->m_health == 0 || misc::is_bit_set((int*)&ped->m_damage_bits, 8))
						continue;

					if (ped->m_health < ped->m_maxhealth) 
					{
						g_pointers->m_give_pickup_rewards(1 << player.second->id(), REWARD_HEALTH);
					}
					if (ped->m_armor < 50.0f) 
                                        {
						g_pointers->m_give_pickup_rewards(1 << player.second->id(), REWARD_ARMOUR);
					}
				}
			}
		}
	}
};

```

**Reply by mentolixite** [reply]
#261

**Comment by iwanttodo1t** [comment]
Teleportation and God modes can make the average player grief
Now yimmenu supports session tracking

**Reply by iwanttodo1t** [reply]
So, whether it causes grief depends on the user

**Reply by Yimura** [reply]
Not sure if you realise this but we're actually working on adding toxic features, YimMenu will no longer attempt to be non-toxic, however certain things will still not be allowed like crashes, infinite loading screens and other things that may completely stop other players games from working.

Please see #641

**Reply by iwanttodo1t** [reply]
Sorry, I dug the grave （Discuss outdated content）

**Reply by Yimura** [reply]
It's fine, let's keep this here for other people that might stumble across this discussion.
